src/consensus/params.h

```diff
@@ -11,6 +11,17 @@
 
 namespace Consensus {
 
+enum BuriedDeployment : int16_t
+{
+    // buried deployments get negative values to avoid overlap with DeploymentPos
+    DEPLOYMENT_HEIGHTINCB = std::numeric_limits<int16_t>::min(),
+    DEPLOYMENT_CLTV,
+    DEPLOYMENT_DERSIG,
+    DEPLOYMENT_CSV,
+    DEPLOYMENT_SEGWIT,
+};
+constexpr bool ValidDeployment(BuriedDeployment dep) { return DEPLOYMENT_HEIGHTINCB <= dep && dep <= DEPLOYMENT_SEGWIT; }
```

Comment:
> very minor nit: If you add a `DEPLOYMENT_BURIED_MAX` and then use `dep < DEPLOYMENT_BURIED_MAX` instead of `dep <= DEPLOYMENT_SEGWIT` then this line wouldn't need to change when something is buried. But I am not sure if it's an improvement overall.

Reply:
> > very minor nit: If you add a `DEPLOYMENT_BURIED_MAX` and then use `dep < DEPLOYMENT_BURIED_MAX` instead of `dep <= DEPLOYMENT_SEGWIT` then this line wouldn't need to change when something is buried. But I am not sure if it's an improvement overall.
> 
>

Reply:
Didn't want to add a `DEPLOYMENT_BURIED_MAX` since that would then need to be included in any `switch` statements.